File: backend/libs/models/src/user/model-inners/supplier-settings.ts
import { prop } from '@typegoose/typegoose'
import { WebsiteLink } from './supplier-link'

export class SupplierSettingsModel {
	@prop({ default: false })
	searchable: boolean

	@prop()
	publicName?: string

	@prop()
	apiName?: string

	@prop({ default: 'link' })
	ymlLoadType: 'link' | 'file'

	@prop()
	ymlLinkSettings?: {
		url: string
		refreshTime: '5m' | '10m' | '30m' | '1h' | '3h' | '12h' | '1d'
	}

	@prop()
	description: string

	@prop()
	links: Array<WebsiteLink>
}

File: backend/Backend.Core/Entities/User/Supplier/SupplierSettings.cs
namespace Backend.Core.Entities.User.Supplier;

public class SupplierSettings : IBaseSettings
{
    public int Id { get; set; }
    
    public int UserId { get; set; }
    
    public string PublicName { get; set; } = null!;
    
    public string ApiName { get; set; } = null!;
    
    public string? Description { get; set; }
    
    public bool Searchable { get; set; }
    
    public int YmlLoadTypeId { get; set; }
    
    public string? YmlLink { get; set; }
   
    public int? RefreshTimeId { get; set; }
    
    public virtual RefreshTime RefreshTime { get; set; } = null!;
    
    public DateTime OffersUpdatedAtUtc { get; set; }
    
    public virtual ICollection<Link> Links { get; set; } = null!;
}